src/aatoolbox/datasources/chirps/chirps.py

```diff
@@ -0,0 +1,631 @@
+                f"{str(self._resolution).replace('.', 'p')}/"
+                ".T/last/subgrid/0./add/T/table%3A/1/%3Atable/.csv"
+            )
+
+            datetime_object = datetime.strptime(
+                pd.read_csv(url).values[0][0], "%d %b %Y"
+            )
+
+        else:  # monthly
+            url = (
+                "https://iridl.ldeo.columbia.edu/SOURCES/.UCSB/.CHIRPS/.v2p0/"
+                ".monthly/.global/"
+                ".T/last/subgrid/0./add/T/table%3A/1/%3Atable/.csv"
+            )
+
+            datetime_object = datetime.strptime(
+                pd.read_csv(url).values[0][0], "%b %Y"
+            )
+            day = calendar.monthrange(
+                datetime_object.year, datetime_object.month
+            )[1]
+            datetime_object = datetime_object.replace(day=day)
+
+        return datetime_object.date()
+
+
+class ChirpsMonthly(Chirps):
+    """
+    .
+
+    Examples
+    --------
+    >>> from aatoolbox import create_country_config, CodAB, ChirpsMonthly
+    >>> country_config = create_country_config(iso3="bfa")
+    >>> codab = CodAB(country_config=country_config)
+    >>> codab.download()
+    >>> admin0 = codab.load(admin_level=0)
+    >>> geo_bounding_box = GeoBoundingBox.from_shape(admin0)
+    >>> chirps_monthly = ChirpsMonthly(country_config=country_config,
+                                    geo_bounding_box=geo_bounding_box)
+    >>> chirps_monthly.download(
+            start_year=2007,
+            end_year=2020,
+            start_month=10,
+            end_month,2
+            )
+    >>> chirps_monthly.process()
+    >>> chirps_monthly_data = chirps_daily.load()
+    """
+
+    if Chirps.__doc__ is not None:
+        __doc__ = Chirps.__doc__ + __doc__
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, frequency="monthly", **kwargs)
+
+
+class ChirpsDaily(Chirps):
+    """
+    .
+
+    Examples
+    --------
+    >>> from aatoolbox import create_country_config, CodAB, ChirpsMonthly
+    >>> country_config = create_country_config(iso3="bfa")
+    >>> codab = CodAB(country_config=country_config)
+    >>> codab.download()
+    >>> admin0 = codab.load(admin_level=0)
+    >>> geo_bounding_box = GeoBoundingBox.from_shape(admin0)
+    >>> chirps_daily = ChirpsDaily(country_config=country_config,
+                                    geo_bounding_box=geo_bounding_box)
+    >>> chirps_daily.download(start_year=2007, start_month=10, start_day=23)
+    >>> chirps_daily.process()
+    >>> chirps_daily_data = chirps_daily.load(
+        start_year=2012,
+        end_year=end_2021,
+        start_month=6,
+        end_month=4
+        )
+    """
+
+    if Chirps.__doc__ is not None:
+        __doc__ = Chirps.__doc__ + __doc__
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, frequency="daily", **kwargs)
+
+
+@check_file_existence
+def _actual_download(filepath: Path, url: str, clobber: bool) -> Path:
```

Comment:
The `@check_file_existence` decorator was improved in #99 such that it can be part of the class. I.e. you can intend `_actual_download` one level and instead call `self._actual_download`